src/main/java/javaserver/Response.java

```diff
@@ -30,10 +31,14 @@ public String getBody() {
 	}
 
 	public String formatResponse() {
-		String output = getResponseCode() + System.lineSeparator();
-		output += getHeader() + System.lineSeparator();
+		String output = statusLine + System.lineSeparator();
+		if (header != null) {
+			output += header + System.lineSeparator();
+		}
 		output += System.lineSeparator();
-		output += getBody() + System.lineSeparator();
+		if (body != null) {
+			output += body + System.lineSeparator();
```

Comment:
May be useful to pull out a helper method here to concatenate a string with `System.lineSeparator()`